### PR TITLE
2022: Update lint rules and fix errors

### PR DESCRIPTION
**Pull request by tylerpalmerdev**
### Ticket(s)

-   [Issue](https://airtable.com/appfJZShN8K4tcWHU/tblXQZfKPAJIjV4cL/viw2LxioXmyNePLBu/recdR5ljrfJn0rWBR?blocks=hide)

### Type of change

_Please delete any options that are not relevant_

-   [x] Bug fix (non-breaking change which fixes an issue)

### Description

Consolidating conflicting lint rules. Prettier gets priority for all formatting. Change a few errors to warnings to be fixed later. Auto-fixed all errors that I could, manually fixed the rest.

### Screenshots
n/a

### Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (if applicable)
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
-   [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/Components/DonateLevelCard.jsx [hunk 1]
```diff
@@ -15,25 +15,23 @@ const useStyles = makeStyles(theme => ({
     },
 }));
 
-const DonateLevelCard = ({ amount, image, text, alt }) => {
+function DonateLevelCard({ amount, image, text, alt }) {
```
Comment (stuttskl) [inline src/Components/DonateLevelCard.jsx:18]: Is there a reason we want to move away from arrow functions?

File: .eslintrc.yaml [hunk 2]
```diff
@@ -17,57 +18,37 @@ overrides:
...
+    - "**/*.spec.js"
+    - "build/"
+settings:
+    import/resolver:
+        - typescript: "{}"
 parser: "@typescript-eslint/parser"
 parserOptions:
     ecmaFeatures:
-        jsx: true
+        jsx: false
```
Comment (stuttskl) [inline .eslintrc.yaml:30]: Are we saying we do not want to be able to use jsx components?

**Review by stuttskl — COMMENTED** [review]
Please see comments : )

File: src/Components/DonateLevelCard.jsx (hunk 1 above)
Comment (IsaiahPetrichor) [reply src/Components/DonateLevelCard.jsx:18]: I also prefer arrow functions 👍

File: .prettierrc.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "tabWidth": 4,
     "semi": true,
     "singleQuote": false,
-    "printWidth": 120,
+    "printWidth": 100,
```
[the remark this answers is not in the document]
Comment (stuttskl) [reply .prettierrc.json:6]: Yeahhhh, I generally like AirBnb's style guide, but I've used 100 as printwidth in a project at work and I found that more often than not it cut lines off in weird places. My preference would be 120, but also fine with trying 100..

File: src/Components/DonateLevelCard.jsx (hunk 1 above)
Comment (stuttskl) [reply src/Components/DonateLevelCard.jsx:18]: I see. I think I'd prefer to use function components that still embrace arrow functions and be using ES6 styntax

**Review by stuttskl — APPROVED** [review]

**Pull request merged**